### PR TITLE
[SER-449] Added test for baseline serialization round trip

### PR DESCRIPTION
**Pull request by TheButlah**
Does not include functionality to test Strings or Handles. Will add that when I add support for serializing Handles.

### CODE REVIEW

File: client/rust/src/contract/mod.rs [hunk 1]
```diff
@@ -85,8 +85,8 @@ impl ContractData {
     }
 }
 
-#[rsharp::substitute("tp_client::contract")]
 #[cfg(feature = "c_api")]
+#[rsharp::substitute("tp_client::contract")]
```
Comment (tomthecarrot) [inline client/rust/src/contract/mod.rs:89]: Why was this moved? Is it style or does it affect the functionality?

Resolve when answered.

File: client/rust/src/lib.rs [hunk 2]
```diff
@@ -15,7 +15,6 @@
     while_true
 )]
 #![allow(clippy::diverging_sub_expression)]
-#![feature(generic_associated_types)]
```
Comment (tomthecarrot) [inline client/rust/src/lib.rs:18]: Woot! #stable

Resolve when seen.

**Review by tomthecarrot — APPROVED** [review]
Tested locally - passed test!

File: client/rust/src/contract/mod.rs (hunk 1 above)
Comment (TheButlah) [reply client/rust/src/contract/mod.rs:89]: style, I preferred the cfg gate at the top. Technically attribute order can also affect functionality, I believe the outermost attribute executes first. But for a cfg macro it disables not just the item its on but all the attribute macros anyway, so it doesn't matter here.
